File: array/sort_one_n.py
arr = [3, 2, 4, 1, 5]

i = 0
while i < len(arr):
    correct = arr[i] - 1
    if correct == i:
        i += 1
    else:
        arr[i], arr[correct] = arr[correct], arr[i]

print(arr)

File: array/sort_one_n.js
var arr = [3, 4, 2, 5, 1];

var i = 0;
while (i < arr.length) {
    let correct = arr[i] - 1;
    if (i === correct) {
        i ++;
    } else {
        [arr[i], arr[correct]] = [arr[correct], arr[i]];
    }
}

console.log(arr);
